specs/agents/common-public-apis.md

```diff
@@ -0,0 +1,31 @@
+## Common Public APIs
+
+This document describes APM Agent Public APIs that are or should be common
+between the separate language implementations; and that are not defined in
+a separate document such as the [Tracing API](./tracing-api.md).
+
+In the following APIs, the string "apm" is used to refer to the top-level
+namespace or primary singleton API object provided by the APM agent:
+
+- Node.js: the singleton [Agent instance](https://www.elastic.co/guide/en/apm/agent/nodejs/current/agent-api.html)
+- Go: the [Tracer](https://www.elastic.co/guide/en/apm/agent/go/current/api.html#tracer-api) instance
+- Java: the [`co.elastic.apm.api.ElasticApm` class](https://www.elastic.co/guide/en/apm/agent/java/current/public-api.html)
+- .NET: the [`Elastic.Apm.Agent` class](https://www.elastic.co/guide/en/apm/agent/dotnet/current/public-api.html)
+- Python: the [`elasticapm` package](https://www.elastic.co/guide/en/apm/agent/python/current/api.html)
+- Ruby: the [`ElasticAPM` module](https://www.elastic.co/guide/en/apm/agent/ruby/current/api.html)
+- RUM JS: the singleton [Agent instance](https://www.elastic.co/guide/en/apm/agent/js-base/current/agent-api.html)
+
+
+### `apm.getServiceName() -> String`
```

Comment:
For Java, the API is an optional dependency and thus, we can't assume it's available even if the agent is installed. To set the service name in ECS logging, the agent has to instrument the ECS logger's constructors.